src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -512,14 +512,14 @@ protected function compileStatements($value)
     protected function compileStatement($match)
     {
         if (str_contains($match[1], '@')) {
-            $match[0] = isset($match[3]) ? $match[1].$match[3] : $match[1];
+            $match[0] = isset($match[3]) ? $match[1].$match[3] : $match[1].$match[2];
         } elseif (isset($this->customDirectives[$match[1]])) {
             $match[0] = $this->callCustomDirective($match[1], Arr::get($match, 3));
         } elseif (method_exists($this, $method = 'compile'.ucfirst($match[1]))) {
             $match[0] = $this->$method(Arr::get($match, 3));
         }
 
-        return isset($match[3]) ? $match[0] : $match[0].$match[2];
```

Comment:
Can you full explain this change?

Reply:
I'll try...

For the string `@media only screen and (min-width:480px) {`

The regex in the `compileStatements()` function returns the following matches...

```
^ array:3 [
  0 => "@media "
  1 => "media"
  2 => " "
]
```

`matches[0]` contains the matched directive including a trailing space. This was originally returned along with the space that was matched in `$matches[2]`. Infact, all we actually need to do in this instance is return `$matches[0]`. None of the `if` statements in the `compileStatement` function match, so the directive is just returned as-is, which is exactly what we want.

I `dd()`'d all the `$matches` across the entire test suite to see in which scenarios we actually need to return the extra trailing space.

It seems that the only time the trailing space in `matches[2]` needs to be returned is when a directive has been escaped...

The following string `@@continue` outputs the following matches...

```
^ array:3 [
  0 => "@@continue "
  1 => "@continue"
  2 => " "
]
```

Here, the string we actually want to output is `$matches[1]`, but that does not contain the trailing space, so we need to add that from `$matches[2]`, The first `if` statement in the `compileStatement` function now handles this.

If you need any further explanation, please let me know 👍